driven/storage/migrations.go

```diff
@@ -0,0 +1,1127 @@
+						break
+					}
+				}
+
+				if !groupExists {
+					if appMemberships[index].Groups == nil {
+						appMemberships[index].Groups = make([]accountGroup, 0)
+					}
+					appMemberships[index].Groups = append(appMemberships[index].Groups, group)
+				}
+			}
+
+			// merge preferences
+			for k, v := range appMembership.Preferences {
+				if _, foundKey := appMemberships[index].Preferences[k]; !foundKey {
+					appMemberships[index].Preferences[k] = v
+				}
+			}
+			// tenant accounts existing before this migration have no stored secrets (nothing to merge)
+
+			// use the newer of the two most recent client versions
+			if appMemberships[index].MostRecentClientVersion == nil {
+				appMemberships[index].MostRecentClientVersion = appMembership.MostRecentClientVersion
+			} else if appMembership.MostRecentClientVersion != nil {
+				existingClientVersionNumbers := model.VersionNumbersFromString(*appMemberships[index].MostRecentClientVersion)
+				clientVersionNumbers := model.VersionNumbersFromString(*appMembership.MostRecentClientVersion)
+
+				if clientVersionNumbers != nil && !clientVersionNumbers.LessThanOrEqualTo(existingClientVersionNumbers) {
+					versionString := clientVersionNumbers.String()
+					appMemberships[index].MostRecentClientVersion = &versionString
+				}
+			}
+		}
+	}
+
+	return appMemberships
+}
+
+func (sa *Adapter) migrateLoginSessions(context TransactionContext, removedAuthTypes map[string]model.AuthType) error {
+	// remove the following fields from all login sessions
+	update := bson.D{primitive.E{Key: "$unset", Value: bson.D{
+		primitive.E{Key: "account_auth_type_id", Value: 1},
+		primitive.E{Key: "account_auth_type_identifier", Value: 1},
+		primitive.E{Key: "external_ids", Value: 1},
+	}}}
+	res, err := sa.db.loginsSessions.UpdateManyWithContext(context, bson.M{}, update, nil)
+	if err != nil {
+		return errors.WrapErrorAction(logutils.ActionUpdate, model.TypeLoginSession, nil, err)
+	}
+	if res.ModifiedCount != res.MatchedCount {
+		return errors.ErrorAction(logutils.ActionUpdate, model.TypeLoginSession, &logutils.FieldArgs{"matched": res.MatchedCount, "modified": res.ModifiedCount})
+	}
+
+	for oldCode, authType := range removedAuthTypes {
+		// update the auth_type_code field for all removed auth types
+		update := bson.D{
+			primitive.E{Key: "$set", Value: bson.D{
+				primitive.E{Key: "auth_type_code", Value: authType.Code},
+			}},
+		}
+
+		res, err := sa.db.loginsSessions.UpdateManyWithContext(context, bson.M{"auth_type_code": oldCode}, update, nil)
+		if err != nil {
+			return errors.WrapErrorAction(logutils.ActionUpdate, model.TypeLoginSession, &logutils.FieldArgs{"auth_type_code": oldCode}, err)
+		}
+		if res.ModifiedCount != res.MatchedCount {
+			return errors.ErrorAction(logutils.ActionUpdate, model.TypeLoginSession, &logutils.FieldArgs{"auth_type_code": oldCode, "matched": res.MatchedCount, "modified": res.ModifiedCount})
+		}
+	}
+
+	return nil
+}
+
+// migrate to tenants accounts
```

Comment:
We already removed all this code for tenants accounts migration as it was deployed on all environments.

Reply:
I added these changes for the migration back in to perform the migration in the ROKMETRO database. I realize I made a mistake in changing the `"migrated_2"` field name to `"migrated"`, which would cause the migration to run again on accounts in the ROKWIRE database, so I fixed that issue.

Reply:
Why don't we remove all this migration code related with the tenants accounts? Or this is not applied into the Rokmetro databases yet?

Reply:
Yes, the tenant account migration has not yet been applied to the Rokmetro databases.

Reply:
Hi @roberlander2. I understand but I think we should merge into the Rokwire code only the new stuff you have developed and changed but not old Rokwire code. In the latest code in Rokwire the tenant  accounts migration code does not exist. It was removed with this issue https://github.com/rokwire/core-building-block/issues/698 We had issues with the tenants accounts migration and we reverted the version the first time. Finally we fixed it and all now in Rokwire is working as expected and the migration code was removed. Maybe you should get the latest code state from Rokwire develop and merge into your branch. As you have not deployed this yet then maybe better to put this tenant migration code alive but only in the Rokmetro world not introducing it again in Rokwire with your PR. In addition to your huge PR including couple issues handled, some refactoring and your migration stuff, keeping also the tenants migration code complicates the transfer of your code into the Rokwire code and poses a risk for issues. Let me know if you have any comments. Thanks.

Reply:
Thanks @petyos, I think you are right that we should not reintroduce the tenant accounts migration, so I removed it.

Reply:
Thanks @roberlander2